src/poetry/installation/executor.py

```diff
@@ -300,10 +300,12 @@ def _execute_operation(self, operation: Operation) -> None:
                     io = self._sections.get(id(operation), self._io)
 
                 with self._lock:
-                    trace = ExceptionTrace(e)
-                    trace.render(io)
                     pkg = operation.package
-                    if isinstance(e, IsolatedBuildError):
+                    with_trace = True
+
+                    if isinstance(e, IsolatedBuildBackendError):
+                        # we disable trace here explicitly to workaround incorrect context detection by crashtest
```

Comment:
@Secrus since you have been handling cleo, this seems to be needed since internally it does not respect `__suppress_context__` when a parent exception exists.

Reply:
:thinking: would you mind submitting a ticket to Cleo's issue tracker? I will investigate.

Reply:
Also, put a `# TODO` or `# FIXME` here so we don't loose that.

Reply:
Sure, I'll see if I can get a minimal reproducer.

Reply:
See https://github.com/python-poetry/cleo/issues/454

Reply:
Added `TODO`.